core/variant/variant.h

```diff
@@ -738,6 +738,7 @@ class Variant {
 	static bool has_utility_function_return_value(const StringName &p_name);
 	static Variant::Type get_utility_function_return_type(const StringName &p_name);
 	static bool is_utility_function_vararg(const StringName &p_name);
+	static bool is_utility_function_const(const StringName &p_name); // For GDScript. Compile time constant. This is **not** `const` qualifier.
```

Comment:
I prefer `pure`, personally.
I think it would be better to unmark functions that may be problematic as constant for this time. Another PR can be created to further discuss about these functions and mark some of them as constant when necessary.

Reply:
The problem is that `pure` means very clear and strict [requirements](https://en.wikipedia.org/wiki/Pure_function):

1. the function return values are identical for identical arguments,
2. the function has no side effects.

For this reason, we cannot consider `str()` pure, since for objects it calls `_to_string()` but its purity is unknown.